apps/judicial-system/backend/src/app/modules/event-log/models/eventLog.model.ts

```diff
@@ -18,12 +18,39 @@ import { Case } from '../../case/models/case.model'
   timestamps: false,
 })
 export class EventLog extends Model {
-  static caseSentToCourtEvent(eventLogs?: EventLog[]) {
-    return eventLogs?.find(
-      (eventLog) =>
-        eventLog.eventType === EventType.CASE_SENT_TO_COURT ||
-        eventLog.eventType === EventType.INDICTMENT_CONFIRMED,
-    )
+  static getEventLogByEventType(
+    eventType: EventType | EventType[],
+    eventLogs: EventLog[] | undefined,
+    userRole?: UserRole,
+  ): EventLog | undefined {
+    if (!eventLogs) {
+      return undefined
+    }
+
+    const eventTypes = Array.isArray(eventType) ? eventType : [eventType]
+    const relevantLogs = eventLogs
+      .filter(
+        (eventLog) =>
+          eventTypes.includes(eventLog.eventType) &&
+          (!userRole || eventLog.userRole === userRole),
```

Comment:
What is the purpose of userRole in the context of events? is is access control or just to indicate that a given event is only relevant for a certain role?

Reply:
Pretty certain is it the latter as I also answered to myself above 😁 "It is just because we have the samt event type for several roles and we use it as a way to distinguish a given type 👍"